src/libs/actions/IOU.ts

```diff
@@ -2314,6 +2327,23 @@ function calculateDiffAmount(iouReport: OnyxEntry<OnyxTypes.Report>, updatedTran
     return 0;
 }
 
+function calculateAmountForUpdatedWaypoint(transaction: OnyxEntry<OnyxTypes.Transaction>, transactionChanges: TransactionChanges, policy: OnyxEntry<OnyxTypes.Policy>) {
+    let updatedAmount: number = CONST.IOU.DEFAULT_AMOUNT;
+    let updatedMerchant = Localize.translateLocal('iou.fieldPending');
+    if (transactionChanges?.routes) {
+        const customUnitRateID = TransactionUtils.getRateID(transaction) ?? '';
+        const mileageRates = DistanceRequestUtils.getMileageRates(policy);
+        const mileageRate = mileageRates?.[customUnitRateID];
+        const {unit, rate} = mileageRate;
+        const distance = TransactionUtils.getDistance(transaction);
+        updatedAmount = -DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate ?? 0);
+        updatedMerchant = DistanceRequestUtils.getDistanceMerchant(true, distance, unit, rate, transaction?.currency ?? CONST.CURRENCY.USD, Localize.translateLocal, (digit) =>
+            toLocaleDigit(preferredLocale, digit),
+        );
+    }
+    return {updatedAmount, updatedMerchant};
```

Comment:
NAB: Maybe we could return amount, modifiedAmount, and modifiedMerchant directly so we can just spread the result into the updatedTransaction?

Reply:
Look good to me.